src/types/name.jl

```diff
@@ -0,0 +1,56 @@
+struct SName
+    region::ShortString15
+    locality1::ShortString15
+    locality2::ShortString15
+end
```

Comment:
Maybe should be a subtype of `AbstractString`?

Reply:
I was considering it.
But it is a lot of work to subtype AbstractString. Especially because we need to match equality and thus hash with `String`.
And this is a internal type that is used for an internal field.
And the main operation that matters is comparing for equality with other `SName`s

Reply:
Why not just use a larger `ShortString` and skip this custom type all together? If `ShortString63` is larger than you want you could always tweak the size by defining a primitive type and use `ShortString{T}`. This gets you the string functionality without you having to write it.

Reply:
If you think that is viable than sure.
Based on your earlier comments that i reinterated here https://github.com/JuliaTime/TimeZones.jl/issues/271#issuecomment-816623708
I thought it wasn't
In particular due to the BitInterger seralization one (which i now realized has been fixed).

I also though that they would be slower, as i have found that they are much slower than expected once you get large, but it seems that that doesn't really kick in for ShortString63.
It is a bit slower for some operations but  a bit faster for others.

## Benchmarks

### == false
```julia
julia> @btime x==y setup=(x=convert(TimeZones.SName, "America/Winnipeg"); y=convert(TimeZones.SName, "America/Argentina/ComodRivadavia"))
  1.504 ns (0 allocations: 0 bytes)
false

julia> @btime x==y setup=(x=ShortString63("America/Winnipeg"); y=ShortString63("America/Argentina/ComodRivadavia"))
  2.117 ns (0 allocations: 0 bytes)
false
```

### == true
(basically exactly the same as the false case)
```julia

julia> @btime x==y setup=(x=convert(TimeZones.SName, "America/Winnipeg"); y=convert(TimeZones.SName, "America/Winnipeg"))
  1.505 ns (0 allocations: 0 bytes)
true

julia> @btime x==y setup=(x=ShortString63("America/Winnipeg"); y=ShortString63("America/Winnipeg"))
  2.115 ns (0 allocations: 0 bytes)
true
```

### hash
```julia

julia> @btime hash(x)==hash(y) setup=(x=convert(TimeZones.SName, "America/Winnipeg"); y=convert(TimeZones.SName, "America/Argentina/ComodRivadavia"))
  34.256 ns (0 allocations: 0 bytes);
  
julia> @btime hash(x)==hash(y) setup=(x=ShortString63("America/Winnipeg"); y=ShortString63("America/Argentina/ComodRivadavia"));
  27.598 ns (0 allocations: 0 bytes)
```


So I will make that change, since it is simpler